src/routes/v1/productRoutes.js

```diff
@@ -17,7 +21,7 @@ router.post(
   catchAsync(productController.create)
 );
 
-router.get("/:id", catchAsync(productController.findById));
+router.put("/:id", isProductAuthor, uploadImage, productController.edit);
```

Comment:
Why are we using put here PUT method signifies the whole payload is getting replaced with the new one, which is not the case here, for partial modification PATCH makes more sense.

Reply:
The way we are updating is by pre-populating the edit form with the current properties of the product (as shown below)
<img width="708" alt="image" src="https://github.com/sabeelhps/CB-Jan23-E-Commerce/assets/61115629/02595bbe-7b94-4be2-ba73-78375597438e">

User can then make change to one or more fields and the whole product object is sent on form submission with changed and unchanged properties and replaces the whole product resource in db. For eg, changed quantity to 400 from 4 and remaining fields unchanged. On form submission complete product object would be sent
<img width="627" alt="image" src="https://github.com/sabeelhps/CB-Jan23-E-Commerce/assets/61115629/706da373-4346-4909-9325-e0bef83b6f3c">

Reply:
Based on the UX, we are updating the whole resource and hence using PUT

I can add PATCH Rest endpoint for updating the product as well

Reply:
But properties such as reviews array, productAuthor are constant i believe, we can use patch here

Reply:
Also why are we passing rating field here???

Reply:
agreed...will remove rating from edit form
agreed...product author will remain constant in our app.
for reviews, there is a separate endpoint /reviews.

In this edit form, the product owner will have capability to change the core properties of product itself and we are pre-populating with the existing data so that product owner can see what previous value was before making the change.

Reply:
@sabeelhps I have changed the PUT to PATCH endpoint and made the required changes in the repo and service level as well.